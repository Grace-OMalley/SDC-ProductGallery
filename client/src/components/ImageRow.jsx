
import React from 'react';

class ImageRow extends React.Component {

  render() {
    return (
      <ul className="imageRow">
        {this.props.images.map((image, index) => (
          <li onMouseOver={(e) => this.props.mouseHandler(e)} data-index={index} key={index} className={this.props.selected === index ? 'selected' : ''}><img src={image} /></li>
        )
        )}
      </ul>
    );
  }
}

export default ImageRow;