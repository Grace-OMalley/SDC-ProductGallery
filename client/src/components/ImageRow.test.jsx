import React from 'react';
import renderer from 'react-test-renderer';
import ImageRow from './ImageRow';

it('renders correctly', () => {
  const tree = renderer
    .create(<ImageRow images={['Image1', 'Image2', 'Image3']} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});