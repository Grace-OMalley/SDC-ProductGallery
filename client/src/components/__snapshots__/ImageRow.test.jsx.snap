// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="imageRow"
>
  <li
    className=""
    data-index={0}
  >
    <img
      src="Image1"
    />
  </li>
  <li
    className=""
    data-index={1}
  >
    <img
      src="Image2"
    />
  </li>
  <li
    className=""
    data-index={2}
  >
    <img
      src="Image3"
    />
  </li>
</ul>
`;
